File: parallelepiped.cpp
//------------------------------------------------------------------------------
// parallelepiped.cpp - содержит функции обработки параллелепипеда
//------------------------------------------------------------------------------

#include "parallelepiped.h"

//------------------------------------------------------------------------------
// Ввод параметров параллелепипеда из потока
void In(parallelepiped &p, ifstream &ifst) {
    ifst >> p.a >> p.b >> p.c;
}

// Случайный ввод параметров параллелепипеда
void InRnd(parallelepiped &p) {
    p.a = Random();
    p.b = Random();
    p.c = Random();
    p.density = RandomDouble();
}

//------------------------------------------------------------------------------
// Вывод параметров параллелепипеда в поток
void Out(parallelepiped &p, ofstream &ofst) {
    ofst << "It is Parallelepiped: a = "
         << p.a << ", b = " << p.b
         << ", c = " << p.c
         << ". SurfaceArea = " << SurfaceArea(p) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление площади поверхности параллелепипеда
double SurfaceArea(parallelepiped &p) {
    return 2 * (p.a * p.b + p.b * p.c + p.a * p.c);
}

File: avs3/parallelepiped.py
import random


# ----------------------------------------------
# Ввод параметров параллелепипеда из файла.
def in_file(parallelepiped, line):
    str_array = line.split(' ')
    parallelepiped.append("parallelepiped")
    parallelepiped.append(float(str_array[1]))  # density.
    parallelepiped.append(int(str_array[2]))  # a.
    parallelepiped.append(int(str_array[3]))  # b.
    parallelepiped.append(int(str_array[4]))  # c.
    parallelepiped.append(surface_area(parallelepiped))  # Surface area.


# Случайный ввод параметров параллелепипеда.
def in_rnd(parallelepiped):
    parallelepiped.append("parallelepiped")
    parallelepiped.append(round(random.uniform(0.1, 20.0), 3))  # density.
    parallelepiped.append(random.randint(1, 20))  # a.
    parallelepiped.append(random.randint(1, 20))  # b.
    parallelepiped.append(random.randint(1, 20))  # c.
    parallelepiped.append(surface_area(parallelepiped))  # Surface area.


# Вычисление плозади поверхности параллелепипеда.
def surface_area(parallelepiped):
    return float(2.0 * (parallelepiped[2] * parallelepiped[3] +
                        parallelepiped[2] * parallelepiped[4] +
                        parallelepiped[3] * parallelepiped[4]))
